src/common/command_buffers/details/clear.hpp

```diff
@@ -169,4 +169,147 @@ namespace commandbuffers
   public:
     int stencil;
   };
+
+  // Base class for ClearBuffer commands that use vector values
+  template <typename Derived, CommandBufferType Type, typename ValueType>
+  class ClearBufferBaseRequest : public TrCommandBufferSimpleRequest<Derived, Type>
+  {
+  public:
+    ClearBufferBaseRequest() = delete;
+    ClearBufferBaseRequest(int buffer, int drawbuffer, const std::vector<ValueType> &values)
+        : TrCommandBufferSimpleRequest<Derived, Type>()
+        , buffer(buffer)
+        , drawbuffer(drawbuffer)
+        , values(values)
+    {
+    }
+    ClearBufferBaseRequest(const ClearBufferBaseRequest &that, bool clone = false)
+        : TrCommandBufferSimpleRequest<Derived, Type>(that, clone)
+        , buffer(that.buffer)
+        , drawbuffer(that.drawbuffer)
+        , values(that.values)
+    {
+    }
+
+    TrCommandBufferMessage *serialize() override
+    {
+      auto message = TrCommandBufferSimpleRequest<Derived, Type>::serialize();
+      if (values.size() > 0)
+        message->addVecSegment(values);
+      return message;
+    }
+
+    void deserialize(TrCommandBufferMessage &message) override
+    {
+      TrCommandBufferSimpleRequest<Derived, Type>::deserialize(message);
+
+      auto valuesSegment = message.nextSegment();
+      if (valuesSegment != nullptr)
+        values = valuesSegment->toVec<ValueType>();
+      else
+        values.clear();
+    }
+
+    std::string toString(const char *line_prefix) const override
+    {
+      std::stringstream ss;
+      ss << TrCommandBufferSimpleRequest<Derived, Type>::toString(line_prefix) << "("
+         << buffer << ", " << drawbuffer << ", [" << values.size() << " values])";
+      return ss.str();
+    }
+
+  public:
+    int buffer;
+    int drawbuffer;
+    std::vector<ValueType> values;
+  };
+
+  class ClearBufferfvCommandBufferRequest final
+      : public ClearBufferBaseRequest<ClearBufferfvCommandBufferRequest,
+                                      COMMAND_BUFFER_CLEAR_BUFFERFV_REQ,
+                                      float>
+  {
```

Comment:
If there is no extra fields in the child class, we can use `using` instead of writing the similar constructors again, see: https://github.com/M-CreativeLab/jsar-runtime/blob/main/src/common/command_buffers/details/uniforms.hpp#L815-L819